### DISCUSSION TITLE
themes

### DISCUSSION THREAD

**Discussion by bryant-the-coder** [nvim-lualine/lualine.nvim#511] [Q&A]
how can i change the icon in different mode, changing the theme according to the time

**Comment by shadmansaleh** [comment]
Sorry I have no clue what you're talking about . Specify what you're looking for properly please.

**Reply by bryant-the-coder** [reply]
hmm... when i m in insert mode i want it to be a different icons same goes to visual,command etc. When the time is for example 8pm it change to tokyodark. 7am gruvbox.

**Reply by bryant-the-coder** [reply]
is it possible?

**Reply by shadmansaleh** [reply]
1. What icon do you want to change ?

2. If you set theme to `auto` it will update lualine theme as you change colorscheme .

**Reply by bryant-the-coder** [reply]
> 1. What icon do you want to change ?
> 
> 2. If you set theme to `auto` it will update lualine theme as you change colorscheme .

when in command mode, it will be a terminal icon. when in insert, it will be a pencil

2. sorry for the confusion. i meant that when time change my nvim theme change. like 8am gruvbox, 7pm tokyonight storm style

**Reply by shadmansaleh** [reply]
1. In that case just use a custom component for mode like https://github.com/nvim-lualine/lualine.nvim/blob/9e26823ea6c7361aba3253c8a5c56a6f35b4a0ee/examples/evil_lualine.lua#L91-L117 . Instead of changing colors change the icon.
```lua
lualine_a = {function()
   local mode_icons = {
      n = 'n',
      i = 'i',
      v = 'v',
     -- ..... Set icons for the modes you want . 
  }
  return mode_icons[vim.fn.mode()] or 'default'
  end
}
```
2. If you want it on entire nvim not lualine it's **`off topic`** here .
  but you can adapt the code from @fitrh to execute `:colorscheme` command (`vim.cmd('colorscheme color_name')`) instead of calling lualine.setup and put it in your init.vim. 

Also I'd recommend you to checkout neovim's [discource](https://neovim.discourse.group) & [reddit](https://www.reddit.com/r/neovim) forums . There you can find more people that can help you in this case

**Reply by bryant-the-coder** [reply]
does it include custom coloring?

**Comment by fitrh** [comment]
> how can i change the icon in different mode

What icon you want to change ?

> changing the theme according to the time

You can do something like this *(if you are using `packer.nvim`)*

```lua
use({
    "nvim-lualine/lualine.nvim",
    config = function()
        local hours = tonumber(os.date("%-H"))
        local theme = "gruvbox"
        if hours >= 19 or hours < 7 then
            theme = "tokyonight"
        end
        require("lualine").setup({
            options = {
                theme = theme,
                -- the rest of your options
            },
            -- the rest of your lualine configuration
        })
    end,
})
```

**Reply by bryant-the-coder** [reply]
what if i dont want it for lualine as i think i said something wrong lol. i want it for everything. where can i place it??

**Reply by fitrh** [reply]
I assume you have this in your packer configuration
```lua
use({
  "folke/tokyonight.nvim",
  config = function()
    -- your config
  end,
  module = "tokyonight",
})

use({
  "marko-cerovac/material.nvim",
  config = function()
    -- your config
  end,
  module = "material",
})

```

then in yout init file (*e.g. `init.lua`*) after calling packer, put something like this

```lua
local hours = tonumber(os.date("%-H"))
local theme = "material"
if hours >= 19 or hours < 7 then
    theme = "tokyonight"
end
require(theme)
```

**Reply by fitrh** [reply]
If you want to change the mode component icon base on current mode, take a look at my custom
[mode component](https://github.com/fitrh/init.nvim/blob/main/lua/plugin/lualine/component/mode.lua), just replace the `text` for every mode with your icon

**Reply by bryant-the-coder** [reply]
i need to put that in my init.lua? cannot other folder??

**Reply by fitrh** [reply]
Of course you can, just make sure that you put them after the packer loaded

**Reply by bryant-the-coder** [reply]
Error:
``` lua
 ✗ Failed to move tokyonight.nvim: C:\Users\User\AppData\Local\nvim-data\site\pack\packer\start\tokyonight.nvim → C:\Users\User\AppData\Local\nvim-data\site\pack\packer\opt\tokyonight.nvim
```

Packer conf:
``` lua
local fn = vim.fn
local install_path = fn.stdpath('data')..'/site/pack/packer/start/packer.nvim'
if fn.empty(fn.glob(install_path)) > 0 then
  packer_bootstrap = fn.system({'git', 'clone', '--depth', '1', 'https://github.com/wbthomason/packer.nvim', install_path})
end

return require('packer').startup(function()
    -- Packer can manage itself
    use "wbthomason/packer.nvim"

    -- Dependencies
    use "nvim-lua/plenary.nvim"
    use "kyazdani42/nvim-web-devicons"

    -- Theme
    use "tiagovla/tokyodark.nvim"
    use({
      "folke/tokyonight.nvim",
      module = "tokyonight",
    })
    use "morhetz/gruvbox"
    use "drewtempelmeyer/palenight.vim"
    use "arcticicestudio/nord-vim"
    use {"rose-pine/neovim", as = "rose-pine"}

    -- Explorer menu
    use {"kyazdani42/nvim-tree.lua"}

    -- Treesitter
    use {
        "nvim-treesitter/nvim-treesitter",
        run = ":TSUpdate",
    }
    use "p00f/nvim-ts-rainbow"
    use "nvim-treesitter/playground"
    use "windwp/nvim-ts-autotag"
    use "windwp/nvim-autopairs"

    -- Statusline
    use "nvim-lualine/lualine.nvim"

    -- Telescope
    use {"nvim-telescope/telescope.nvim"}

    -- Dashboard
    use "glepnir/dashboard-nvim"

    -- Bufferline
    use "akinsho/bufferline.nvim"

    -- Colorizer
    use "norcalli/nvim-colorizer.lua"

    -- LSP
    use "neovim/nvim-lspconfig"
    use "hrsh7th/nvim-cmp"
    use "hrsh7th/cmp-nvim-lsp"
    use "hrsh7th/cmp-buffer"
    use "hrsh7th/cmp-path"
    use "onsails/lspkind-nvim"
    use "folke/lsp-colors.nvim"
    use "David-Kunz/cmp-npm"
    use "hrsh7th/cmp-vsnip"
    use "hrsh7th/vim-vsnip"

    -- Indentation
    use "lukas-reineke/indent-blankline.nvim"

    -- Auto-comment
    use "numToStr/Comment.nvim"
    use "JoosepAlviste/nvim-ts-context-commentstring"

    -- Terminal
    use "akinsho/toggleterm.nvim"

    -- Harpoon
    use "ThePrimeagen/harpoon"

    -- Disturbance free writing
    use {
    "folke/twilight.nvim",
    "folke/zen-mode.nvim"
    }

    -- Git
    use "lewis6991/gitsigns.nvim"

end)
```

**Reply by bryant-the-coder** [reply]
now it give another error
``` lua
Error detected while processing C:\Users\User\AppData\Local\nvim\init.lua:
E5113: Error while calling lua chunk: C:\Users\User\AppData\Local\nvim/lua/theme.lua:48: Vim(lua):E5108: Error executing lua [string ":lua"]:1: module 'tokyonight' not found:
        no field package.preload['tokyonight']
        no file '.\tokyonight.lua'
        no file 'C:\Users\User\scoop\apps\neovim\current\bin\lua\tokyonight.lua'
        no file 'C:\Users\User\scoop\apps\neovim\current\bin\lua\tokyonight\init.lua'
        no file '.\tokyonight.dll'
        no file 'C:\Users\User\scoop\apps\neovim\current\bin\tokyonight.dll'
        no file 'C:\Users\User\scoop\apps\neovim\current\bin\loadall.dll'
stack traceback:
        [C]: in function 'require'
        [string ":lua"]:1: in main chunk
        [C]: in function 'cmd'
        C:\Users\User\AppData\Local\nvim/lua/theme.lua:48: in main chunk
        [C]: in function 'require'
        C:\Users\User\AppData\Local\nvim\init.lua:4: in main chunk
stack traceback:
        [C]: in function 'cmd'
        C:\Users\User\AppData\Local\nvim/lua/theme.lua:48: in main chunk
        [C]: in function 'require'
        C:\Users\User\AppData\Local\nvim\init.lua:4: in main chunk
```

**Reply by fitrh** [reply]
Where did you set the theme ?

**Reply by fitrh** [reply]
I think this is load ordering problem, try to creat `plugin` directory *(**NOT** inside `lua`)* and then create a file called `theme.lua` and put the logic to load the theme in there

**Reply by fitrh** [reply]
So you have a structur like this

```txt
│   init.lua
│
├───lua
│       plugin.lua
│       your_lua_modules\
│
└───plugin
        packer_compiled.lua
        theme.lua
```

**Reply by bryant-the-coder** [reply]
my folder structure is right here 

https://github.com/bryant-the-coder/NvHack

**Reply by bryant-the-coder** [reply]
but your conf doesnt say so... hmmm

**Reply by fitrh** [reply]
You can follow the above structure

**Reply by fitrh** [reply]
> but your conf doesnt say so... hmmm

Yeah, because I don't use these kind of configuration, but I have tested what I tell
you in another machine

**Reply by fitrh** [reply]
![Screenshot (24)](https://user-images.githubusercontent.com/55179750/147641351-35e9c9af-4b5c-4d2c-b73b-7b86a75d4245.png)

This is a new configuration that I have tested on a windows machine

**Reply by bryant-the-coder** [reply]
y is your windows machine looks so nice?

**Reply by bryant-the-coder** [reply]
btw i got it working
